Upgrade React Native `0.71.11`

fluiddot opened the pull request:
## Related PRs
* [Gutenberg](https://github.com/WordPress/gutenberg/pull/51303)
* [WordPress-Android](https://github.com/wordpress-mobile/WordPress-Android/pull/18613)
* [WordPress-iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/20956)

<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

**To test:**
Follow testing instructions from https://github.com/WordPress/gutenberg/pull/51303.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Upgrade `@babel` dependencies

fluiddot committed:
Upgrade `metro-react-native-babel-preset` and `metro-resolver` dependencies

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Upgrade `metro-react-native-babel-preset` and `metro-resolver` dependencies

This upgrade is needed to support React Native version `0.71.11`.

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'trunk' into upgrade/react-native-0.71.8

# Conflicts:
#	gutenberg

fluiddot committed:
Fix issue with UUID library in Jest

This fixes an issue related to the UUID library when running tests. Without this change, we get the following error:
```
/gutenberg-mobile/gutenberg/node_modules/uuid/dist/esm-browser/index.js:1
({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export { default as v1 } from './v1.js';
                                                                                      ^^^^^^

    SyntaxError: Unexpected token 'export'

      3 |  */
      4 | import fastDeepEqual from 'fast-deep-equal/es6';
    > 5 | import { v4 as uuid } from 'uuid';
        | ^
      6 |
      7 | /**
      8 |  * WordPress dependencies
```

fluiddot committed:
Mock API requests in VideoPress tests

fluiddot committed:
Fix `generateFetchMocks` util import

dcalhoun commented on jest.config.js:
```diff
@@ -56,6 +56,9 @@ module.exports = {
 		'test/helpers$': '<rootDir>/' + configPath + '/helpers.js',
 		jetpackConfig:
 			'<rootDir>/jetpack/tools/js-tools/jest/jest-jetpack-config.js',
+		// Workaround for Jest not having ESM support yet
+		// Reference: https://t.ly/9ap_
+		uuid: require.resolve( 'uuid' ),
```
From reviewing the URL in this comment, my understanding is that we can remove this workaround when we [upgrade to `uuid@9`](https://github.com/uuidjs/uuid/issues/451#issuecomment-1206284480). Is that accurate? It would likely be beneficial to note this in the inline comment.

fluiddot replied:
Ah, true. I haven't tested this myself, but based on the comments that seems the case. I'll update the inline comment to reflect this. Thanks 🙇 !

fluiddot replied:
I tried using `uuid` version 9.x but I still had to use this workaround to address the issue. There might be something else producing the issue, especially as it's not happening in Gutenberg unit tests which I assume we use an older version `uuid` 🤔 .

dcalhoun commented on src/test/videopress/edit.js:
```diff
@@ -35,14 +36,21 @@ import {
+	} );
 	registerJetpackBlocks( DEFAULT_PROPS );
+
+	// Mock request reponses
+	setupApiFetch( generateFetchMocks() );
```
Why this now required due to the RN upgrade? Does this relate to https://github.com/facebook/react-native/pull/34659?

fluiddot replied:
Yep, it looks related to that change. Without this change, the tests fail with the following error:
```
TypeError: Cannot read property 'then' of undefined

      53 | 			method: 'POST',
      54 | 			body: fetchParams.body,
    > 55 | 		} ).then( response => {
         | 		   ^
      56 | 			const { metadata_token } = response;
      57 | 			if ( ! metadata_token ) {
      58 | 				console.warn( 'Token is not achievable' ); // eslint-disable-line no-console
```

I understand that when `Promise` was polyfilled in Jest tests, we didn't need to explicitly mock API responses (i.e. `api-fetch` library).

dcalhoun committed:
build: Upgrade @react-native-community/cli to ^1.10.0

dcalhoun committed:
build: Upgrade eslint to ^8.19.0

fluiddot committed:
Update Gutenberg ref

dcalhoun committed:
build: Catpure iOS XCFramework Podfile changes

Capture result of `bundle exec pod install` within the `ios-xcframework`
directory. while the XCFramework is not currently used for WordPress iOS
integration, this is required to resolve CI failures.

fluiddot committed:
Fix VideoPress block test cases (#5916)

* Add util to generate VideoPress privacy settings for tests

* Mock VideoPress check ownership API request

* Update VideoPress edit test cases

Instead of using a constant for the block HTML, now we generate the HTML via the `generateBlockHTML` util.

* Use different fetch mocks in VideoPress edit test cases

This is needed to keep the same block attributes set via the block HTML generation.

Extract fetch mock metadata to a constant

* Update VideoPress description setting placeholder query

* Update VideoPress test snapshots

The only change is that the video is now public when setting a caption. This has no impact in the testing logic.

* Update src/test/videopress/local-helpers/utils.js

Fixes the JSDoc type of `isVideoPrivate` and `isSitePrivate` params.

Co-authored-by: David Calhoun <github@davidcalhoun.me>

* Fix value used to generate privacy setting in `generatePrivacySettings`

* Remove unneeded metadata on fetch mocks

---------

Co-authored-by: David Calhoun <github@davidcalhoun.me>

fluiddot committed:
Revert "build: Upgrade eslint to ^8.19.0"

This reverts commit 0cdf825c9da412e3b3b0fc182c25f0e9eae12b8c.

fluiddot committed:
Fix `isPrivate` value used in `generatePrivacySettings` function

fluiddot committed:
Fix ESLint issues in VideoPress test utils

fluiddot committed:
Upgrade React Native `0.71.11` - Android changes (#5872)

* build: Update Gutenberg ref

* build: Capture package-lock.json changes

The result of running:

```
rm -rf ./node_modules && npm install
```

* Update Gutenberg ref

* Update a11y id queries for Android E2E tests

---------

Co-authored-by: David Calhoun <github@davidcalhoun.me>

SiobhyB committed:
Upgrade React Native `0.71.11` - iOS changes (#5881)

* Update Gutenberg reference

* Update script for generating React-Codegen

The method we used to generate React-Codegen (get_react_codegen_spec) was moved to a different file in newer versions of React Native. We need to update the script for generating it accordingly.

* Update third-party podspecs to target 0.71.11

* Update FBReactNativeSpec.podspec.json with hash

* Apply bundle changes

* Manually add React-bridging.podspec.json

The file was automatically deleted via the generate-podspecs.sh script. It has been manually added as a temporary measure, to enable further testing.

* Revert "Manually add React-bridging.podspec.json"

This reverts commit 970730516b139aed933d854de7f8cf30b0fdf699.

* Update list of dependencies in Podfile for iOS

* Add tag to Hermes podspec to workaround error

This is intended as a temporary workaround, we should implement a more robust fix before merging.

* Update dependency versions in Podfile.lock

* [Temporary] Add React Native as a dependency

* Revert "[Temporary] Add React Native as a dependency"

This reverts commit 21a4fbed95e8f6e6af7d10addd4a380b79623c8b.

* Update Gutenberg ref

* Update Gutenberg ref

* Update Gutenberg ref

* Disable hermes when generating codegen podspec

* Generate third-party podspecs

* Update FBReactNativeSpec podspec

---------

Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'trunk' into upgrade/react-native-0.71.8

# Conflicts:
#	bundle/ios/App.js
#	bundle/ios/App.js.map
#	gutenberg

fluiddot committed:
Update bundle

fluiddot committed:
Update localization strings

fluiddot committed:
Update Android bundle assets

fluiddot committed:
Enable Hermes in podspec generation

fluiddot committed:
Update React-Codegen podspec

fluiddot committed:
Update FBReactNativeSpec podspec

fluiddot committed:
Update Gutenberg ref

mokagio committed:
Update XCFramework setup to work with React Native 0.71.11 (#5924)

* Update Gutenberg reference

* Update script for generating React-Codegen

The method we used to generate React-Codegen (get_react_codegen_spec) was moved to a different file in newer versions of React Native. We need to update the script for generating it accordingly.

* Update third-party podspecs to target 0.71.11

* Update FBReactNativeSpec.podspec.json with hash

* Apply bundle changes

* Manually add React-bridging.podspec.json

The file was automatically deleted via the generate-podspecs.sh script. It has been manually added as a temporary measure, to enable further testing.

* Revert "Manually add React-bridging.podspec.json"

This reverts commit 970730516b139aed933d854de7f8cf30b0fdf699.

* Update list of dependencies in Podfile for iOS

* Add tag to Hermes podspec to workaround error

This is intended as a temporary workaround, we should implement a more robust fix before merging.

* Update dependency versions in Podfile.lock

* [Temporary] Add React Native as a dependency

* Revert "[Temporary] Add React Native as a dependency"

This reverts commit 21a4fbed95e8f6e6af7d10addd4a380b79623c8b.

* Update Gutenberg ref

* Update Gutenberg ref

* Point RNReanimated to a custom branch to work around 2.17.0 issue

The issue is due to RNReanimated looking for react-native in the
node_modules relative to its location, which is how most projects would
be setup. Unfortunately, our projects is not standard.

When running `pod install` or `pod update`, we'd get:

```
Installing RNReanimated 2.17.0 (was 2.9.1-wp-4)
internal/modules/cjs/loader.js:934
  throw err;
  ^

Error: Cannot find module 'react-native/package.json'
Require stack:
- /Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/[eval]
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:931:15)
    at Function.resolve (internal/modules/cjs/helpers.js:113:19)
    at [eval]:1:9
    at Script.runInThisContext (vm.js:134:12)
    at Object.runInThisContext (vm.js:310:38)
    at internal/process/execution.js:81:19
    at [eval]-wrapper:6:22
    at evalScript (internal/process/execution.js:80:60)
    at internal/main/eval_string.js:27:3 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/[eval]'
  ]
}

[!] Invalid `RNReanimated.podspec` file: no implicit conversion of nil into String.

 #  from /var/folders/dq/cdqxvx3s5ps75564rpmb_dc00000gn/T/d20230627-89428-abwplk/RNReanimated.podspec:5
 #  -------------------------------------------
 #  reanimated_package_json = JSON.parse(File.read(File.join(__dir__, "package.json")))
 >  config = find_config()
 #  assert_no_multiple_instances(config)
 #  -------------------------------------------
```

* Print message during `pod install` about `REACT_NATIVE_NODE_MODULES_DIR`

* Setup NVM and node modules for RNReanimated in XCFramework CI step

* Apply bundle changes

* Switch to using `use_react_native!` in XCFramework `Podfile`

Using only our custom specs, the build failed with

```
Multiple commands produce '/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/DerivedData/XCFrameworkScaffold/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/CallbackWrapper.h'

- Target 'ReactCommon' (project 'Pods') has copy command from '/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/Pods/ReactCommon/react/bridging/CallbackWrapper.h' to '/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/DerivedData/XCFrameworkScaffold/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/CallbackWrapper.h'
- Target 'ReactCommon' (project 'Pods') has copy command from '/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/Pods/ReactCommon/react/nativemodule/core/ReactCommon/CallbackWrapper.h' to '/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/DerivedData/XCFrameworkScaffold/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/CallbackWrapper.h'
```

However, with this setup the build fails with:

```
/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/Pods/../../gutenberg/node_modules/react-native/React/FBReactNativeSpec/../../scripts/xcode/with-environment.sh: line 35: .xcode.env: command not found
/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/Pods/../../gutenberg/node_modules/react-native/React/FBReactNativeSpec/../../scripts/xcode/with-environment.sh: line 35: node: command not found
[Warning] You need to configure your node path in the  environment.  You can set it up quickly by running:  echo 'export NODE_BINARY=' > .xcode.env  in the ios folder. This is needed by React Native to work correctly.  We fallback to the DEPRECATED behavior of finding . This will be REMOVED in a future version.  You can read more about this here: https://reactnative.dev/docs/environment-setup#optional-configuring-your-environment
/Users/gio/Developer/a8c/gutenberg-mobile/ios-xcframework/Pods/../../gutenberg/node_modules/react-native/React/FBReactNativeSpec/../../scripts/xcode/with-environment.sh: line 41: /scripts/find-node-for-xcode.sh: No such file or directory
[Error] Could not find node. It looks like that the .xcode.env or .xcode.env.local
```

* Add React Native `post_install` hook

* Use `.xcode.env` instead of `.xcode.env.local`

This way, users can override what CocoaPods does without modifying the
codebase via the `.local` file.

* Bump XCFramework deployment target to iOS 15.0

* Use Xcode 14.3.1 in CI, like WordPress iOS does

Otherwise, we're getting a compilation failure.
See
https://github.com/wordpress-mobile/WordPress-iOS/commit/eadad98d81c8970144707d8967f8c9f06a6ada38#commitcomment-119894859

* Disable Hermes when building XCFramework

* Refine messaging when setting `REACT_NATIVE_NODE_MODULES_DIR` env var

* Set `SKIP_INSTALL = NO` at the `xcconfig` level, too

* Switch an `unless !=` to `if ==`

* Add workaround to setup Hermes correctly in XCFramework

They will take place if the process runs with `HERMES_ENABLED` true.

* Enable building XCFramework with Hermes

* Embed Hermes XCFramework in Gutenberg XCFramework

This is so that, at runtime, apps can find it. Otherwise, the build will
run but the app will fail at runtime with:

```
dyld[99988]: Library not loaded: @rpath/hermes.framework/hermes
  Referenced from: <931F58EC-3AC7-39DE-BEB7-1D44D8A5CF15> /Users/gio/Library/Developer/CoreSimulator/Devices/90F54127-B854-4ED8-93FA-D58167331941/data/Containers/Bundle/Application/3DA26EC7-B308-4029-9974-426839535542/WordPress.app/Frameworks/Gutenberg.framework/Gutenberg
```

* Prevent XCFramework build script from deleting RN-code-generated files

* Update Gutenberg ref

* Set `REACT_NATIVE_NODE_MODULES_DIR` env var before fetching Reanimated pod

* Update `Podfile.lock` file

* Add `.xcode.env` file with dynamic node binary path

---------

Co-authored-by: Siobhan <siobhan@automattic.com>
Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

fluiddot committed:
Generate iOS bundle with bytecode

fluiddot committed:
Update bundle (using bytecode)

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

mokagio committed:
WIP – Link against Hermes, but do not embed it

The idea being to avoid nested frameworks because of embedding and
instead adding Hermes as a another vendored framework in the podspec.

mokagio committed:
WIP – Use Hermes directly from the RN pods when creating the tar.gz

mokagio committed:
Do not copy the 1GB worth of Catalyst Hermest XCFramework

fluiddot committed:
Update release notes

fluiddot committed:
Merge branch 'upgrade/react-native-0.71.8' into mokagio/use-hermes-from-cocoapods

# Conflicts:
#	gutenberg

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'trunk' into upgrade/react-native-0.71.8

# Conflicts:
#	__device-tests__/gutenberg-editor-sanity-test-1-visual.test.js
#	bundle/ios/App.js
#	bundle/ios/App.js.map
#	gutenberg
#	ios-xcframework/Podfile.lock

fluiddot committed:
Fix entry in release notes

fluiddot committed:
Merge branch 'upgrade/react-native-0.71.8' into mokagio/use-hermes-from-cocoapods

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'trunk' into upgrade/react-native-0.71.8

# Conflicts:
#	gutenberg

SiobhyB reviewed (approved):
Approved via https://github.com/WordPress/gutenberg/pull/51303#pullrequestreview-1537530236.

For posterity, noting that https://github.com/wordpress-mobile/gutenberg-mobile/pull/5973 needs to be merged in for the tests to pass, and to avoid the issue with crashing on iOS.

mokagio committed:
Do not embed Hermes in the Gutenberg XCFramework (#5973)

fluiddot commented:
> For posterity, noting that #5973 needs to be merged in for the tests to pass, and to avoid the issue with crashing on iOS.

The PR has been already merged in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5874/commits/378e2330385ca3279f60f4f04b2683d405e3c1e5.

SiobhyB committed:
Merge branch 'trunk' into upgrade/react-native-0.71.8

SiobhyB committed:
Update Gutenberg reference

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'trunk' into upgrade/react-native-0.71.8

# Conflicts:
#	gutenberg

SiobhyB committed:
Update Gutenberg reference

SiobhyB committed:
Update appender button's XPATH for Android

With this change, the most up-to-date details for the XPATH are now in place.

SiobhyB committed:
Ensure top column block is selected before removal

SiobhyB committed:
Simplify XPATH for appender button selection

This commit significantly simplifies the XPath used to identify the column block's appender button in the Android application. Previous long and potentially unstable XPath has been replaced with a more concise and reliable one, reducing the dependence on the complete UI hierarchy and improving the maintainability and stability of the automation script.

SiobhyB committed:
Merge pull request #5986 from wordpress-mobile/fix/sanity-test-1-errors

RN upgrade: Fix test suite 1 errors

dcalhoun committed:
Merge branch 'trunk' of github.com:wordpress-mobile/gutenberg-mobile into upgrade/react-native-0.71.8

dcalhoun committed:
build: Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref with merge commit

fluiddot committed:
Update release notes

fluiddot committed:
Remove deprecated inline comment

We decided to go with Hermes, so we no longer need to mention that we are still deciding.

fluiddot committed:
Fix iOS bundle folder path creation

fluiddot committed:
Update XCFramework `Podfile.lock` file

fluiddot commented:
As a side note, https://github.com/wordpress-mobile/gutenberg-mobile/pull/5874/commits/b837251232ac5ec1240bd852cda176e872e9b074 was needed because I updated `Podfile` in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5874/commits/23d9fbd0e729d45823cae7a90c8d35625d2a67be 🤦.
